### DISCUSSION TITLE
About controlling application

### DISCUSSION THREAD

**Discussion by 10100010010001001010** [axmolengine/axmol#1272] [General]
What do you think, is it worth moving the implementation of the control to a separate file, or is it better to create it right away in the init() function of the scene? How do you add controls to your project?

**Comment by DelinWorks** [comment]
use components, you can add that component to any scene you'd like to have controls [see this example.](https://github.com/DelinWorks/Rebound/blob/custom-editor-support/Source/Components/SceneInputManagerComponent.cpp)

**Reply by 10100010010001001010** [reply]
What about button bindings? In your project, you implement them right on the scene.cpp. What do you think, is it worth putting them in a separate file or is it a waste of time?

**Reply by DelinWorks** [reply]
no honestly i didn't feel the need to put them in a separate file, and I honestly don't know anything about biding controls but I guess it's simple, just a map of int to int bidings and the value is what binding you change in the map, and for each key press you check for what binding it is there and execute it's function
